File: src/main/java/kr/co/lunatalk/domain/payment/repository/PaymentRepository.java
package kr.co.lunatalk.domain.payment.repository;

import kr.co.lunatalk.domain.payment.domain.Payment;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.Optional;

public interface PaymentRepository extends JpaRepository<Payment, Long> {

	Optional<Payment> findByOrder_OrderNumber(String orderNumber);

	Optional<Payment> findByPaymentKey(String paymentKey);
}

File: src/main/kotlin/kr/co/lunatalk/domain/payment/repository/PaymentRepository.kt
package kr.co.lunatalk.domain.payment.repository

import kr.co.lunatalk.domain.payment.domain.Payment
import org.springframework.data.jpa.repository.JpaRepository
import java.util.Optional

interface PaymentRepository : JpaRepository<Payment, Long>, PaymentRepositoryCustom {

    fun findByOrder_OrderNumber(orderNumber: String): Optional<Payment>

    fun findByPaymentKey(paymentKey: String): Optional<Payment>
}
